packages/react-router/src/router.ts

```diff
@@ -2183,8 +2194,6 @@ export class Router<
     } catch (err) {
       if (isRedirect(err)) {
         return await this.preloadRoute({
-          fromSearch: next.search,
```

Comment:
What's the purpose of removing `from` in the route preload?

Or is this automatically being resolved in `buildLocation`?

Reply:
It seems like router was using the preloaded location for `from`. Problem with this is it might not match the current location and can cause the invariant to trigger. We decided its better if the user explicitly passes `from` if they want relative pathing